[OSDOCS#18992] CQA pre-migration for Windows Containers index assembly (#115719)
openshift/openshift-docs: modules/managing-windows-container-workloads.adoc

// Module included in the following assemblies:
//
// * windows_containers/index.adoc
//
// This file is a navigation file - xrefs are allowed in this type of module, but it must only ever be included from one location

:_mod-docs-content-type: REFERENCE
[id="managing-windows-container-workloads_{context}"]
= Managing Windows container workloads

[role="_abstract"]
With a Red Hat subscription, you can get support for running Windows workloads in {product-title}.

Windows instances deployed by the WMCO are configured with the containerd container runtime. For more information, see the xref:../windows_containers/wmco_rn/windows-containers-release-notes.adoc#windows-containers-release-notes[release notes].

> You can add Windows nodes either by creating a xref:../windows_containers/creating_windows_machinesets/creating-windows-machineset-aws.adoc#creating-windows-machineset-aws[compute machine set] or by specifying existing Bring-Your-Own-Host (BYOH) Window instances through a xref:../windows_containers/byoh-windows-instance.adoc#byoh-windows-instance[config map].

Reviewer: ```suggestion
You can add Windows nodes either by creating a xref:../windows_containers/creating_windows_machinesets/creating-windows-machineset-aws.adoc#creating-windows-machineset-aws[compute machine set] or by specifying existing Bring-Your-Own-Host (BYOH) Windows instances through a xref:../windows_containers/byoh-windows-instance.adoc#byoh-windows-instance[config map].
```

> You can add Windows nodes either by creating a xref:../windows_containers/creating_windows_machinesets/creating-windows-machineset-aws.adoc#creating-windows-machineset-aws[compute machine set] or by specifying existing Bring-Your-Own-Host (BYOH) Window instances through a xref:../windows_containers/byoh-windows-instance.adoc#byoh-windows-instance[config map].

Reviewer: ```suggestion
You can add Windows nodes either by creating a xref:../windows_containers/creating_windows_machinesets/creating-windows-machineset-aws.adoc#creating-windows-machineset-aws[compute machine set] or by specifying existing Bring-Your-Own-Host (BYOH) Window instances through a xref:../windows_containers/byoh-windows-instance.adoc#byoh-windows-instance[ConfigMap].
```

You need the WMCO to run Windows workloads in your cluster. The WMCO orchestrates the process of deploying and managing Windows workloads on a cluster. For more information, see xref:../windows_containers/enabling-windows-container-workloads.adoc#enabling-windows-container-workloads[how to enable Windows container workloads].

You can create a Windows `MachineSet` object to create infrastructure Windows machine sets and related machines so that you can move supported Windows workloads to the new Windows machines. You can create a Windows `MachineSet` object on multiple platforms.

You can xref:../windows_containers/scheduling-windows-workloads.adoc#scheduling-windows-workloads[schedule Windows workloads] to Windows compute nodes.

You can xref:../windows_containers/windows-node-upgrades.adoc#windows-node-upgrades[perform Windows Machine Config Operator upgrades] to ensure that your Windows nodes have the latest updates.

You can xref:../windows_containers/removing-windows-nodes.adoc#removing-windows-nodes[remove a Windows node] by deleting a specific machine.

> You can xref:../windows_containers/byoh-windows-instance.adoc#byoh-windows-instance[use Bring-Your-Own-Host (BYOH) Windows instances] to repurpose Windows Server VMs and bring them to {product-title}. BYOH Windows instances benefit users who are looking to mitigate major disruptions if a Windows server goes offline. You can use BYOH Windows instances as nodes on {product-title} 4.8 and later versions.

Reviewer: ```suggestion
You can xref:../windows_containers/byoh-windows-instance.adoc#byoh-windows-instance[use Bring-Your-Own-Host (BYOH) Windows instances] to repurpose Windows Server VMs and bring them to {product-title}. BYOH Windows instances benefit users who are looking to mitigate major disruptions when a Windows server goes offline. You can use BYOH Windows instances as nodes on {product-title} 4.8 and later versions.
```
